src/loki_server.py

```diff
@@ -65,21 +65,17 @@ def version(self) -> str:
             exception is raised
         """
         try:
-            # Un-comment the following lines once new Loki 2.3.1 version is released.
-            #
-            # We are hardcoding version here because there is and bug in last stable Loki version
-            # (2.3.0) that do not return its version: https://github.com/grafana/loki/issues/4133
-            # This bug it was already addressed in PR: https://github.com/grafana/loki/pull/4135
-            # but it's note released yet.
-            #  info = self._build_info()
-            #  return = info.get("version", None)
-            _ = self._build_info()
-            return "2.3.0"
+            info = self._build_info()
+            version = info.get("version", None)
+            if not version:
+                raise LokiServerNotReadyError("Loki version could not be retrieved.")
```

Comment:
Only reason I can think that the version will not be available is that the Loki server is not yet responding to queries. If this is true then I would expect that the Loki charm sets it status to waiting status or defer and event handler depending on the use case for which version is being retrieved. I am not too familiar with the whole charm so I am leaving this to your better judgement.

Reply:
Yes, that's why we raise LokiServerNotReadyError